File: E-Tour/Java/Backend/E-Tour/src/main/java/com/example/models/CostMaster.java
package com.example.models;

import jakarta.persistence.*;

@Entity
public class CostMaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CostId")
    private int costId;

    @Column(name = "TwinSharingcost", nullable = false)
    private int twinSharingCost;

    @Column(name = "extraPersonCost", nullable = false)
    private int extraPersonCost;

    @Column(name = "childWithBed", nullable = false)
    private int childWithBed;

    @Column(name = "childWitoutBed", nullable = false) // Fixed spelling mistake
    private int childWithoutBed;

    @Column(name = "singlePersonCost")
    private int singlePersonCost;

    @OneToOne
    @JoinColumn(name = "tourId", nullable = false)
    private Tours tour;

    // Default Constructor
    public CostMaster() {}

    // Getters and Setters
    public int getCostId() {
        return costId;
    }

    public void setCostId(int costId) {
        this.costId = costId;
    }

    public int getTwinSharingCost() {
        return twinSharingCost;
    }

    public void setTwinSharingCost(int twinSharingCost) {
        this.twinSharingCost = twinSharingCost;
    }

    public int getExtraPersonCost() {
        return extraPersonCost;
    }

    public void setExtraPersonCost(int extraPersonCost) {
        this.extraPersonCost = extraPersonCost;
    }

    public int getChildWithBed() {
        return childWithBed;
    }

    public void setChildWithBed(int childWithBed) {
        this.childWithBed = childWithBed;
    }

    public int getChildWithoutBed() {
        return childWithoutBed;
    }

    public void setChildWithoutBed(int childWithoutBed) {
        this.childWithoutBed = childWithoutBed;
    }

    public int getSinglePersonCost() {
        return singlePersonCost;
    }

    public void setSinglePersonCost(int singlePersonCost) {
        this.singlePersonCost = singlePersonCost;
    }

    public Tours getTour() {
        return tour;
    }

    public void setTour(Tours tour) {
        this.tour = tour;
    }
}

File: Models/Costmaster.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace E_Tour.Models;

[Table("costmaster")]
[Index("TourId", Name = "UK1qg4m2fi7wc915v9six8ivvn2", IsUnique = true)]
public partial class Costmaster
{
    [Key]
    public int CostId { get; set; }

    public int TwinSharingcost { get; set; }

    [Column("childWithBed")]
    public int ChildWithBed { get; set; }

    [Column("childWitoutBed")]
    public int ChildWitoutBed { get; set; }

    [Column("extraPersonCost")]
    public int ExtraPersonCost { get; set; }

    [Column("tourId")]
    public int TourId { get; set; }

    [Column("singlePersonCost")]
    public int  SinglePersonCost { get; set; }

    [ForeignKey("TourId")]
    [InverseProperty("Costmaster")]
    public virtual Tour Tour { get; set; } = null!;

    [InverseProperty("TotalAmountNavigation")]
    public virtual Transactionsstatus? Transactionsstatus { get; set; }
}
